### DISCUSSION TITLE
How to use bluetooth keyboard in dracut?

### DISCUSSION THREAD

**Discussion by EncodedBird** [dracutdevs/dracut#1618] [Q&A] [answered]
I saw from [this PR](https://github.com/dracutdevs/dracut/pull/1139) that you can add a bluetooth module to your initramfs from dracut. I was able to confirm bluetooth was getting added into my initramfs. However, how do I get it to automatically connect a bluetooth keyboard so that I can enter a luks passphrase?

**Comment by mrvantage** [comment, marked as answer]
I had the same problem, that it was not clear how to get the bluetooth module working. It supposed to work automatically when a bluetooth keyboard is detected. The problem I had that dracut was not configured as hostonly, so to make it work I added the following to my dracut configuration:
```
hostonly=yes
```

**Comment by mrvantage** [comment]
To add to this, since of today I have to explicitly enable bluetooth to enable it in dracut:
```
add_dracutmodules+=" bluetooth "
```

**Reply by theking2** [reply]
where do I add this?

**Comment by zearp** [comment]
Sorry to post in an old topic, did not want to create a new one for what is likely user error but I keep getting this warning despite having included it. Maybe not explicitly? How does one add it explicitly?

```
[zearp@mini ~]$ sudo dracut -f
[sudo] password for zearp: 
dracut[W]: If you need to use bluetooth, please include it explicitly.
[zearp@mini ~]$ cat /etc/dracut.conf.d/bt.conf
add_dracutmodules+=" bluetooth "
```
